.github/workflows/db.yml

```diff
@@ -0,0 +1,256 @@
+  push:
+    branches:
+      - '*'         # matches every branch that doesn't contain a '/'
+      - '*/*'       # matches every branch containing a single '/'
+      - '**'        # matches every branch
+    tags:
+      - "*"
+
+  # See: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+env:
+  EXTRA_PHP_EXTENSIONS: intl bcmath zip pcntl exif curl gd pdo_pgsql pdo_mysql pdo_sqlite
+  DOCKER_PHP_EXTENSION_INSTALLER_VERSION: 2.2.2
+  COMPOSER_VERSION: 2.7.1
+
+jobs:
+  migrations-mysql:
+    name: MySQL / MariaDB
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+      # See: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs
+      matrix:
+        php_version:
+          - 8.3
+        db_version:
+          - mariadb:latest
+          - mariadb:11
+          - mariadb:10
+
+          - mysql:latest
+          - mysql:8.3
+          - mysql:8.0
+
+    container:
+      image: php:${{ matrix.php_version }}-cli
+
+    services:
+      db:
+        image: ${{ matrix.db_version }}
+        env:
+          MARIADB_ROOT_PASSWORD: password
+          MARIADB_PASSWORD: password
+          MARIADB_USER: pixelfed
+          MARIADB_DATABASE: pixelfed
+
+          MYSQL_ROOT_PASSWORD: "password"
+          MYSQL_USER: "pixelfed"
+          MYSQL_PASSWORD: "password"
+          MYSQL_DATABASE: "pixelfed"
+        ports:
+          - 33306:3306
+
+      redis:
+        image: redis:7.2
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    # See: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-concurrency-and-the-default-behavior
+    concurrency:
+      group: db-migrations-${{ github.ref }}-${{ matrix.php_version }}-${{matrix.db_version}}
+      cancel-in-progress: true
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Download mlocati/docker-php-extension-installer
+        uses: robinraju/release-downloader@v1.9
+        with:
+          repository: mlocati/docker-php-extension-installer
+          tag: ${{env.DOCKER_PHP_EXTENSION_INSTALLER_VERSION}}
+          fileName: install-php-extensions
+
+      - name: Download composer
+        uses: robinraju/release-downloader@v1.9
+        with:
+          repository: composer/composer
+          tag: ${{env.COMPOSER_VERSION}}
+          fileName: composer.phar
+
+      - name: Download jippi/dottie
+        uses: robinraju/release-downloader@v1.9
+        with:
+          repository: jippi/dottie
+          latest: true
+          fileName: "*_amd64.deb"
+
+      - name: Install dottie
+        run: |
+          dpkg -i *.deb
+          rm -f *.deb
+          dottie -v
```

Comment:
Do you need the deb, or can you just download the binary?

Reply:
The deb is just the binary + 2 shell completion scripts, and it ensures the file ends up in the right path and so on, so difference is ~1-2 KB and the deb file is simpler/more flexible going forward.